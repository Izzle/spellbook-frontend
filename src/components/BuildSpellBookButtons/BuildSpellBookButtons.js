import React from 'react';
import { Link } from 'react-router-dom';
import './BuildSpellBookButtons.css';

export default function BuildSpellBookButtons(props) {

    // need to the eventHnadler to work and THEN we route to another page.
    // after the service sends the new PUT request, chain a .then(props.history.push(`/spellbooks/${paramsId}`))
    // <Link onClick={() => props.onSaveSubmit} to={`/spellbook/${props.paramsId}`} className='BuildSpellBookButtons__Link'>
    
    return(
        <div>
            <button onClick={() => props.onSaveSubmit()} type='submit' className='BuildSpellBookButtons__button'>Save Spellbook</button>

            <Link to='/spellbook' className='BuildSpellBookButtons__Link'>
              <button className='BuildSpellBookButtons__button'>This should ask the user if they're sure they want to go back to their spellbook or go to all spellbooks</button>
            </Link>
        </div>
    );

}