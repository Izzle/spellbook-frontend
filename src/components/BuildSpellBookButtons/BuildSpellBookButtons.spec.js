import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import BuildSpellBookButtons from './BuildSpellBookButtons';
import { BrowserRouter } from 'react-router-dom';


describe('<BuildSpellBookButtons />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><BuildSpellBookButtons /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><BuildSpellBookButtons /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});