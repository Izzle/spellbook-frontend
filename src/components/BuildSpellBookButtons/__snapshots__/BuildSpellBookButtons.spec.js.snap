// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuildSpellBookButtons /> renders this UI as expected 1`] = `
<div>
  <button
    className="BuildSpellBookButtons__button"
    onClick={[Function]}
    type="submit"
  >
    Save Spellbook
  </button>
  <a
    className="BuildSpellBookButtons__Link"
    href="/spellbook"
    onClick={[Function]}
  >
    <button
      className="BuildSpellBookButtons__button"
    >
      View All Spellbooks
    </button>
  </a>
</div>
`;
